src/Mutts.jl

```diff
@@ -109,19 +119,22 @@ Return a mutable shallow copy of Mutts object `obj`, whose children are all stil
 branch!(obj::T) where T = branch!(mutts_trait(T), obj)
 function branch!(::MuttsType, obj)
     branchactions(obj)
-    markimmutable!(obj)
+    mark_immutable!(obj)
 
-    obj = copy(obj)
-    obj.__mutt_mutable = true
+    obj = make_mutable_copy(obj)
+    #obj.__mutt_mutable = true
     obj
 end
 
-
+function make_mutable_copy(obj)
+    @warn "Implement some version of `make_mutable_copy` for your type."
+    error()
+end
```

Comment:
My gut says it would be better to just leave this unimplemented by default, a la:
```suggestion
function make_mutable_copy end
```
This is also what Lyndon suggests here in his blog post:
https://www.oxinabox.net/2020/04/19/Julia-Antipatterns.html (under `NotImplemented Exceptions`), though actually I think this example is more okay than the one in his post, since you're not specifying type restrictions on the arguments.

the error message you currently have isn't very helpful because, e.g., you don't specify for _which_ type isn't implemented, and if you get this deep in a stack trace, it's not very helpful.

Reply:
Still, I'm also sympathetic to wanting to provide a nice message like you've done here, so if you do want to keep it like this, maybe you can do it by still throwing a MethodError instead of `error()`?:
```suggestion
function make_mutable_copy(obj)
    @warn "Implement some version of `make_mutable_copy` for your type."
    throw(MethodError(make_mutable_copy, typeof(obj))
end
```
But that's a bit silly, so i'd probably just stick with not implementing this method at all 👍

Reply:
Thanks. Yeah, your initial suggestion makes a lot of sense. Changing to `function make_mutable_copy end`